[Thrift] Compatibility issues

I am trying to get the sockshop example you provided to work on ubuntu using Thrift instead of gRPC. To do that, i am using the plugin you provided in your codebase. However, when building i get the following errors:

    $ go run wiring/main.go -o $BUILD_DIR -w $spec
  
    ...
  
    Step 6/15 : RUN go build -o /cart_proc ./cart_proc
     ---> Running in 049d9596e76f
    # blueprint/goproc/cart_proc/thrift
    cart_proc/thrift/CartService_OTServerWrapperInterface_ThriftServer.go:36:27: undefined: cartservice.NewCartServiceProcessor
    cart_proc/thrift/CartService_conversions.go:204:18: undefined: double
    The command '/bin/sh -c go build -o /cart_proc ./cart_proc' returned a non-zero code: 1
    ERROR: Service 'cart_ctr' failed to build : Build failed

I tried to use Thrift versions 0.19, 0.9, 0.10, 0.11, 0.12, 0.13 and all of them lead to compatibility issues. 
Are you aware of these issues and can you tell me for which Thrift version you built the plugin?

You can reproduce these issues by executing this code on any debian system: https://github.com/DesjardinsRepos/blueprint

--- a/plugins/thrift/thriftcodegen/clientgen.go
+++ b/plugins/thrift/thriftcodegen/clientgen.go
@@ -98,7 +98,7 @@ func New_{{.Name}}(ctx context.Context, serverAddress string) (*{{.Name}}, error
 	return handler, nil
 }
 
-{{$service := .Service.Name -}}
+{{$service := .Service.BaseName -}}
 {{$receiver := .Name -}}
 {{$prefix := .ImportPrefix -}}
 {{- range $_, $f := .Service.Methods }}

--- a/plugins/thrift/thriftcodegen/clientgen.go
+++ b/plugins/thrift/thriftcodegen/clientgen.go
@@ -105,7 +105,7 @@ func New_{{.Name}}(ctx context.Context, serverAddress string) (*{{.Name}}, error
 func (client *{{$receiver}}) {{SignatureWithRetVars $f}} {
 	// Create and marshall the thrift Request object
 	req := &{{$prefix}}.{{$service}}_{{$f.Name}}_Request{}
-	marshall_{{$f.Name}}_req(req, {{ArgVars $f}})
+	marshall_{{$service}}_{{$f.Name}}_req(req, {{ArgVars $f}})
 
 	ctx, cancel := context.WithTimeout(ctx, client.Timeout)
 	defer cancel()
@@ -122,7 +122,7 @@ func (client *{{$receiver}}) {{SignatureWithRetVars $f}} {
 		return
 	}
 
-	{{RetVarsEquals $f}} unmarshall_{{$f.Name}}_rsp(rsp)
+	{{RetVarsEquals $f}} unmarshall_{{$service}}_{{$f.Name}}_rsp(rsp)
 	return
 }
 {{end}}

--- a/plugins/thrift/thriftcodegen/marshallgen.go
+++ b/plugins/thrift/thriftcodegen/marshallgen.go
@@ -43,7 +43,7 @@ package {{.Package}}
 {{ range $_1, $service := .Services -}}
 {{ range $_2, $method := $service.Methods -}}
 // Client-side function to pack {{$service.Name}}.{{$method.Name}} args into a Thrift {{$pkg}}.{{$method.Request.ThriftType.Name}} struct
-func marshall_{{$method.Name}}_req({{$method.MarshallRequest $imports $pkg}}) *{{$pkg}}.{{$method.Request.ThriftType.Name}} {
+func marshall_{{$service.Name}}_{{$method.Name}}_req({{$method.MarshallRequest $imports $pkg}}) *{{$pkg}}.{{$method.Request.ThriftType.Name}} {
 	{{- range $j, $arg := $method.Request.FieldList}}
 	{{$arg.Marshall $imports "" $pkg}}
 	{{- end}}
@@ -51,7 +51,7 @@ func marshall_{{$method.Name}}_req({{$method.MarshallRequest $imports $pkg}}) *{
 }
 
 // Server-side function to unpack {{$service.Name}}.{{$method.Name}} args from a Thrift {{$method.Request.ThriftType.Name}} struct
-func unmarshall_{{$method.Name}}_req(msg *{{$pkg}}.{{$method.Request.ThriftType.Name}}) (
+func unmarshall_{{$service.Name}}_{{$method.Name}}_req(msg *{{$pkg}}.{{$method.Request.ThriftType.Name}}) (
 	{{- range $j, $arg := $method.Request.FieldList}}{{if $j}}, {{end}}{{$arg.Name}} {{$imports.NameOf $arg.SrcType}}{{end -}}
 ) {
 	{{- range $j, $arg := $method.Request.FieldList}}
@@ -61,7 +61,7 @@ func unmarshall_{{$method.Name}}_req(msg *{{$pkg}}.{{$method.Request.ThriftType.
 }
 
 // Server-side function to pack {{$service.Name}}.{{$method.Name}} retvals into a Thrift {{$pkg}}.{{$method.Response.ThriftType.Name}} struct
-func marshall_{{$method.Name}}_rsp({{$method.MarshallResponse $imports $pkg}}) *{{$pkg}}.{{$method.Response.ThriftType.Name}} {
+func marshall_{{$service.Name}}_{{$method.Name}}_rsp({{$method.MarshallResponse $imports $pkg}}) *{{$pkg}}.{{$method.Response.ThriftType.Name}} {
 	{{- range $j, $ret := $method.Response.FieldList}}
 	{{$ret.Marshall $imports "" $pkg}}
 	{{- end}}
@@ -69,7 +69,7 @@ func marshall_{{$method.Name}}_rsp({{$method.MarshallResponse $imports $pkg}}) *
 }
 
 // Client-side function to unpack {{$service.Name}}.{{$method.Name}} retvals from a Thrift {{$method.Response.ThriftType.Name}} struct
-func unmarshall_{{$method.Name}}_rsp(msg *{{$pkg}}.{{$method.Response.ThriftType.Name}}) (
+func unmarshall_{{$service.Name}}_{{$method.Name}}_rsp(msg *{{$pkg}}.{{$method.Response.ThriftType.Name}}) (
 	{{- range $j, $ret := $method.Response.FieldList}}{{if $j}}, {{end}}{{$ret.Name}} {{$imports.NameOf $ret.SrcType}}{{end -}}
 ) {
 	{{- range $j, $ret := $method.Response.FieldList}}

--- a/plugins/thrift/thriftcodegen/servergen.go
+++ b/plugins/thrift/thriftcodegen/servergen.go
@@ -92,7 +92,7 @@ func (handler *{{.Name}}) Run(ctx context.Context) error {
 	return server.Serve()
 }
 
-{{$service := .Service.Name -}}
+{{$service := .Service.BaseName -}}
 {{$receiver := .Name -}}
 {{$prefix := .ImportPrefix -}}
 {{ range $_, $f := .Service.Methods }}

--- a/plugins/thrift/thriftcodegen/servergen.go
+++ b/plugins/thrift/thriftcodegen/servergen.go
@@ -97,13 +97,13 @@ func (handler *{{.Name}}) Run(ctx context.Context) error {
 {{$prefix := .ImportPrefix -}}
 {{ range $_, $f := .Service.Methods }}
 func (handler *{{$receiver}}) {{$f.Name -}}(ctx context.Context, req *{{$prefix}}.{{$service}}_{{$f.Name}}_Request) (*{{$prefix}}.{{$service}}_{{$f.Name}}_Response, error) {
-	{{ArgVarsEquals $f}} unmarshall_{{$f.Name}}_req(req)
+	{{ArgVarsEquals $f}} unmarshall_{{$service}}_{{$f.Name}}_req(req)
 	{{RetVars $f "err"}} := handler.Service.{{$f.Name}}({{ArgVars $f "ctx"}})
 	if err != nil {
 		return nil, err
 	}
 	rsp := &{{$prefix}}.{{$service}}_{{$f.Name}}_Response{}
-	marshall_{{$f.Name}}_rsp(rsp, {{RetVars $f}})
+	marshall_{{$service}}_{{$f.Name}}_rsp(rsp, {{RetVars $f}})
 	return rsp, nil
 }
 {{end}}

--- a/plugins/thrift/thriftcodegen/thriftgen.go
+++ b/plugins/thrift/thriftcodegen/thriftgen.go
@@ -219,7 +219,7 @@ func (b *ThriftBuilder) makeFieldList(vars []gocode.Variable) ([]*ThriftField, e
 
 // Adds a service declaration for the provided golang service interface.
 func (b *ThriftBuilder) AddService(iface *gocode.ServiceInterface) error {
-	serviceDecl := b.newService(iface.Name)
+	serviceDecl := b.newService(iface.BaseName)
 	for _, method := range iface.Methods {
 		argList, err := b.makeFieldList(method.Arguments)
 		if err != nil {

